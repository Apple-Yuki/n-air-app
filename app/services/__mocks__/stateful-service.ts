const m = jest.requireActual('services/stateful-service');

export function mutation() {
  return function(x: any) {
    return x;
  };
}

export const StatefulService = m.StatefulService;
StatefulService.setupVuexStore({ watch: (x: any) => x });

export function __setup(state?: { [serviceName: string]: any }) {
  StatefulService.setupVuexStore({ watch: (x: any) => x, state });
}
