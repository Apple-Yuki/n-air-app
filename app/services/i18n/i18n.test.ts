import test from 'ava';
import * as Proxyquire from 'proxyquire';

function noopDecorator() {
  return function() {};
}

function identity<T>(x: T): T {
  return x;
}

test('get instance', t => {
  require('../stateful-service')
    .StatefulService
    .setupVuexStore({ watch: identity });

  const m = Proxyquire.noCallThru()('./i18n', {
    'services/stateful-service': {
      mutation: noopDecorator,
      '@noCallThru': false
    },
    '../../util/injector': {
      Inject: noopDecorator
    },
    '../../../obs-api': {
      Global: {}
    },
    'services/app': {
      AppService: {}
    }
  });
  t.truthy(m.I18nService.instance);
});
