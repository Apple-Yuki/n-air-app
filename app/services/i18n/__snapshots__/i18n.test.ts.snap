// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load 1`] = `
Object {
  "en-US": "English",
  "ja-JP": "日本語",
}
`;
